shared/actions/chat2/index.js

```diff
@@ -1018,151 +1045,6 @@ const messageEdit = (action: Chat2Gen.MessageEditPayload, state: TypedState) =>
   }
 }
 
-// First we make the conversation, then on success we dispatch the piggybacking action
-const sendToPendingConversation = (action: Chat2Gen.SendToPendingConversationPayload, state: TypedState) => {
-  const tlfName = action.payload.users.join(',')
-  const membersType = RPCChatTypes.commonConversationMembersType.impteamnative
-
-  return Saga.sequentially([
-    // Disable sending more into a pending conversation
-    Saga.put(Chat2Gen.createSetPendingStatus({pendingStatus: 'waiting'})),
-    // Disable searching for more people once you've tried to send
-    Saga.put(Chat2Gen.createSetPendingMode({pendingMode: 'fixedSetOfUsers'})),
-    // Try to make the conversation
-    Saga.call(RPCChatTypes.localNewConversationLocalRpcPromise, {
-      identifyBehavior: RPCTypes.tlfKeysTLFIdentifyBehavior.chatGui,
-      membersType,
-      tlfName,
-      tlfVisibility: RPCTypes.commonTLFVisibility.private,
-      topicType: RPCChatTypes.commonTopicType.chat,
-    }),
-  ])
-}
-
-// Now actually send
-const sendToPendingConversationSuccess = (
-  results: [any, any, RPCChatTypes.NewConversationLocalRes],
-  action: Chat2Gen.SendToPendingConversationPayload
-) => {
-  const conversationIDKey = Types.conversationIDToKey(results[2].conv.info.id)
-  if (!conversationIDKey) {
-    logger.warn("Couldn't make a new conversation?")
-    return
-  }
-
-  // Update conversationIDKey to real one
-  const {sendingAction} = action.payload
-  const updatedSendingAction = {
-    ...sendingAction,
-    payload: {
-      ...sendingAction.payload,
-      conversationIDKey,
-    },
-  }
-
-  // emulate getting an inbox item for the new conversation. This lets us skip having to unbox the inbox item
-  const dummyMeta = Constants.makeConversationMeta({
-    conversationIDKey,
-    participants: I.OrderedSet(action.payload.users),
-    tlfname: action.payload.users.join(','),
-  })
-
-  return Saga.sequentially([
-    // Clear the search
-    Saga.put(Chat2Gen.createExitSearch({canceled: true})),
```

Comment:
i restructured the side effects so we don't have to do these long lists of actions where if any were missing something would be broken. instead we have more side effects and less dependencies